master/internal/api_generic_tasks.go

```diff
@@ -273,3 +274,35 @@ func (a *apiServer) CreateGenericTask(
 		Warnings: pkgCommand.LaunchWarningToProto(warnings),
 	}, nil
 }
+
+func (a *apiServer) PropegateTaskState(ctx context.Context, taskID model.TaskID, state model.TaskState) error {
+	query := fmt.Sprintf(`
+	WITH RECURSIVE cte as (
+		SELECT * FROM tasks WHERE task_id='%s'
+		UNION ALL
+		SELECT t.* FROM tasks t INNER JOIN cte ON t.parent_id=cte.task_id
+	)
+	UPDATE tasks SET task_state='%s' FROM cte WHERE cte.task_id=tasks.task_id;`, taskID, state)
+	_, err := db.Bun().NewRaw(query).Exec(ctx)
+	return err
+}
+
+func (a *apiServer) FindRoot(ctx context.Context, taskID model.TaskID) (model.TaskID, error) {
+	out := struct {
+		taskID model.TaskID `bun:"table:root"`
+	}{}
+	tree_query := db.Bun().NewSelect().TableExpr("tasks t").Join("my_tree on m.task_id=t.parent_id")
+	task_query := db.Bun().NewSelect().Table("tasks").ColumnExpr("task_id, parent_id, task_id as root").Where("parent_id IS NULL").UnionAll(tree_query)
+	err := db.Bun().NewSelect().Model(&out).
+		WithRecursive("my_tree", task_query).
+		Table("my_tree").Column("root").Where("task_id = ?", taskID).Scan(ctx)
+	return out.taskID, err
+}
```

Comment:
i'm going to go on a limb here and say we shouldn't write this one with bun's. just use Bun().NewRaw(). i feel it is less likely to get broken that way. if it needs some dynamism later, we'll deal with bun.